### ISSUE TITLE
CPU Throttle :-|

### ISSUE THREAD

**Issue by nahuelange** [konkor/cpufreq#50]
Hi,
I do not understand the signifcation of CPU Throttle with the orange « :-| » smiley.
Is it due to overheat? My CPU is at 60°C, not so hot. Is it due to other thing ?

Well it seems that the extension is talking about something going wrong… but do not provide any information for common user.

**Comment by konkor** [comment]
@nahuelange CPU Thermal Throttle can be orange if during the whole current session you get CPU Thermal Throttle or red if you have it now, it will show Throttle Speed so.
**Usually, it tells you that your cooling system is week or needs a cleaning/service**
Normal system doesn't have any CPU Thermal Throttle but some cheap/stock hardware can have this issue even on new systems now. It is a nice feature to protect CPU from the overheating. But you have to determinate the problem and fix it to avoid a desoldering of hardware components.

**Comment by konkor** [comment]
@nahuelange The current temperature (60C) is just current temperature after CPU throttle. So It's not always accurate value of the temperature. **The fact of the CPU thermal throttling is more important than any temperature sensors!**
Also your system doesn't work properly after TT cpu/kernel is trying to disable/reduce all possible features like turbo boost or frequencies for some time.
It's not so easy to describe all this in an extension menu. So it's an only warning...
Regards

**Comment by nahuelange** [comment]
Ok, I understand a bit more.
However, what the signification of the number above the warning?

That is weird, if I look the extension warning, I got a CPU Throttling just after booting my laptop. That make no sense because CPU is cold (<50°C)… then I monitor my CPU sensors, and never go up 65°C that shouldn't be so much.

**Comment by konkor** [comment]
The number is the count of missed CPU cycles for cooling CPU.
Temperature sensors are not showing always actual temperature, usually they show an average approximately value.
Tell me more about your hardware. Is it a laptop? Is it old? What CPU and GPU...

**Comment by nahuelange** [comment]
So, the number of missed CPU cycles is usually 10 (sometimes more when I overload my computer, but it's another thing). I don't know if it's significant to show it?

It's a laptop, 1 year old, with an i7-6700HQ and GeForce GTX 950M. The fan was clean when I opened it 1 month ago to add an M.2 disk.

**Comment by konkor** [comment]
Yeah, TT with 10 cycles is nothing but it's anyway deserve a warning to signal about some issues, weak cooling or airflow.
I think it over. Maybe I'll add some timeout in 10-15 minutes. But here is a problem the app doesn't monitor TT. The extension is checking it on the opening only. So it's like it'll show you the warn on 1st opening but it would be gone after a few minutes if there is no dynamic.

**Comment by nahuelange** [comment]
Thanks for all your answers and your great work!
And maybe, an option to reset the throttling warning, to monitor it could be a great idea.

**Comment by konkor** [comment]
@nahuelange Thanks. But you should know even 1 cycle of TT is dropping the performance for while in the kernel and CPU to cool it down and be sure there is no hardware problems...
I can recommend you to turn off Turbo Boost on laptops because it's very aggressive and quick heat your CPU but cooling is weak and can't provide a stability.
So if you want more responsible system you could try to slightly up minimum frequency (10-15%).
Good luck!

**Comment by nahuelange** [comment]
Yep. I'm going to go a big cleaning of the fans.
Thanks again.

**Comment by konkor** [comment]
@nahuelange Maybe air compressor or vacuum cleaner could be enough for not too old PC...
So you can watch a dynamic of changes with extension ;) And if it's getting worse then go to cleaning...

**Comment by nahuelange** [comment]
You were right! I just cleaned up my CPU radiator and fan, now I have no more throttling.
Thanks again!

**Comment by rtfpessoa** [comment]
@konkor I also have this orange/red message all the time in my laptop.
My temperature is always around 50-60 which is quite normal (after I re-pasted the CPU and GPU). In windows it only show throttle when it gets to 90 and the freq goes down (not even happening now when it benchmarks)

I have the i9-8950HK, how can I better debug this? It is weird for the CPU to be throttling. Can it be a problem from the missed CPU cycles not being reliable with turbo boost?

**Comment by konkor** [comment]
@rtfpessoa 1) 50-60 is an average temperature for all cores/threads.
2) Intel says TT is normal. I think it's marketing. They can place higher frequency without worrying about stable work on that frequency.
3) often Intel has weak stock cooling, you could try to replace it with more efficient cooling system.
4) Turbo Boost is higher frequency, power consumption and temperature.

**Comment by rtfpessoa** [comment]
Have been playing with the settings and for some reason after disabling and enabling turbo boost now I don't get it anymore. Weird since yesterday I could never get rid of it.

**Comment by konkor** [comment]
@rtfpessoa it's not so weird. CPU gets hot on the booting OS. Actually, the extension shows TT only once if there is no changes.

**Comment by toverux** [comment]
> CPU Thermal Throttle can be orange if during the whole current session you get CPU Thermal Throttle or red if you have it now, it will show Throttle Speed so.

If I understand correctly, that means that we get the red face when the CPU is actually throttling. Orange is, what, after throttling? What do you mean by "session"?

**Comment by konkor** [comment]
@toverux You are absolutely right. Sometimes people gets TT after some short work like rendering, compiling or just very rare, so they just want to see a RED WARN all the time :) if there is no TT in current time.

**Comment by rotty3000** [comment]
Hey all. So I've got a brand new (2 months old) Dell Precision 5530 laptop equipped with Intel i9-8950HK and while sitting here writing this comment, CPU idling, Loading at <200%, I am constantly in CPU THROTTLE mode. In fact, cpufreq will go critical just refreshing ASF JIRA web page in Firefox :(

Is this normal? Is there something I can do? Surely the cooling is not plugged up yet.

![screenshot from 2019-01-23 19-17-13](https://user-images.githubusercontent.com/146764/51645757-ab605d80-1f43-11e9-9a80-2ea197f42347.png)
![screenshot from 2019-01-23 19-22-40](https://user-images.githubusercontent.com/146764/51645907-4bb68200-1f44-11e9-9aea-464aaef977c2.png)

**Comment by rotty3000** [comment]
ok, so I re-thermal pasted the CPU and GPU following this guide https://www.ultrabookreview.com/14875-fix-throttling-xps-15/ and it helped a lot.
